lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -144,14 +162,41 @@ class DecomposeAtenSoftmaxIntOp : public OpRewritePattern<AtenSoftmaxIntOp> {
     if (!tensorType.hasDtype() || !tensorType.getDtype().isa<mlir::FloatType>())
       return rewriter.notifyMatchFailure(op, "Only support floating type");
 
-    // exp(x)
-    Value exp = rewriter.create<AtenExpOp>(loc, tensorType, self);
-    // sum(exp(x))
-    Value sum = createSumAlongDimension(rewriter, loc, op, exp, dim, /*keepDim=*/true);
-    if (!sum)
+    Value result = getSoftmaxResult(op, tensorType, rewriter);
+    if (!result)
+      return failure();
+    rewriter.replaceOpWithNewOp<TensorStaticInfoCastOp>(op, op.getType(),
+                                                        result);
+    return success();
+  }
+};
+} // namespace
+
+namespace {
+class DecomposeAten_SoftmaxOp : public OpRewritePattern<Aten_SoftmaxOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(Aten_SoftmaxOp op,
+                                PatternRewriter &rewriter) const override {
+    Value self = op.self();
+    BaseTensorType tensorType = self.getType().cast<BaseTensorType>();
+    if (!tensorType.hasDtype() || !tensorType.getDtype().isa<mlir::FloatType>())
+      return rewriter.notifyMatchFailure(op, "Only support floating type");
+    bool halfToFloat;
+    Type finalType;
+    if (matchPattern(op.half_to_float(), m_TorchConstantBool(&halfToFloat))) {
+      finalType = (halfToFloat)
+                      ? tensorType.getWithSizesAndDtype(tensorType.getSizes(),
+                                                        rewriter.getF32Type())
+                      : tensorType;
+    } else {
```

Comment:
nit: prefer early return rather than using else to simplify control flow. 
```
if (!matchPattern(op.half_to_float(), m_TorchConstantBool(&halfToFloat)))
     return rewriter.notifyMatchFailure
```